### PR TITLE
Issue 99

### LINKED ISSUE #99: Able to create panel with all possible conditions combination
When creating panels, all conditions can be blank.

This also related to https://github.com/transist/echidna-rails/issues/76, I think we should remove the "both" gender. The people don't set their gender should can be included in a panel only if the gender condition is blank (which means all gender). This means we apply same rules on all conditions, no special and complexity.

### PR DESCRIPTION
**Pull request by 0x1b2c**
- Introduce "unknown" object for gender, birth_year and city.
- Create more groups to improve performance of query "all" genders or "all" birth years.

Close #99

### CODE REVIEW

**Comment by 0x1b2c** [comment]
Hey @flyerhzm and @simsicon, would you please help to review this pull request? It changes the under-layer data structure a bit aggressively, it would help a lot if you guys have time to review it, thanks!

**Comment by simsicon** [comment]
Sure, I tried to read it this morning, but did not going well, I will try it tomorrow and comment on some questions.

**Comment by 0x1b2c** [comment]
Yeah, please don't be hesitant for ask anything. Also it would be easier to understand the point if you review it sequently by commits.

**Comment by 0x1b2c** [comment]
I've backed up the old database and finished running the migration on server, it actually cost about 3-4 hours.

Please DO NOT deploy develop branch before this pull request be merged, to ensure the database in a consistent state.

File: app/models/person.rb [hunk 1]
```diff
@@ -40,12 +42,21 @@ class Person
...
+
+  # This is a one side relation since MongoDB has a limitation: Size of a
+  # single BSON document can't be bigger than 16MB.
+  # http://docs.mongodb.org/manual/reference/limits/#BSON Document Size
+  #
+  # Size of a BSON::ObjectId is 12 bytes, storing 2M people (current system
+  # already have 2M+ people) in one group will cause the group document reach
+  # that limitation.
+  # http://docs.mongodb.org/manual/reference/object-id/#overview
+  has_and_belongs_to_many :groups, inverse_of: nil, index: true
```
Comment (simsicon) [inline app/models/person.rb:54]: Good to know this, but can we call group.people if there's no has_and_belongs_to many :people in group.rb?

**Comment by simsicon** [comment]
I think it's good to ship.

**Pull request closed without merging**